eventcore-types/src/projection.rs

```diff
@@ -278,6 +360,31 @@ pub trait EventReader {
         &self,
         after_position: StreamPosition,
     ) -> impl Future<Output = Result<Vec<(E, StreamPosition)>, Self::Error>> + Send;
+
+    /// Read events matching a query, optionally after a given position.
+    ///
+    /// Returns a vector of tuples containing the event and its global position.
+    /// Events are filtered according to the query criteria. If a position is
+    /// provided, only events after that position are returned.
+    ///
+    /// # Type Parameters
+    ///
+    /// - `E`: The event type to deserialize events as
+    ///
+    /// # Parameters
+    ///
+    /// - `query`: The subscription query specifying which events to read
+    /// - `after_position`: Optional position filter (None reads from beginning)
+    ///
+    /// # Returns
+    ///
+    /// - `Ok(Vec<(E, StreamPosition)>)`: Events with their positions
+    /// - `Err(Self::Error)`: If the read operation fails
+    fn read_events_after<E: crate::Event>(
+        &self,
+        query: SubscriptionQuery,
+        after_position: Option<StreamPosition>,
```

Comment:
Is there a reason that we don't put the `after_position` inside the `SubscriptionQuery`? Given that it does, in fact, end up being part of the query?